File: app/channels/chat_channel.rb
# app/channels/chat_channel.rb
class ChatChannel < ApplicationCable::Channel
  def subscribed
    # 固定のストリーム名 "chat_channel" に購読（実際はルームごとに変えることも可能）
    stream_from "chat_channel"
  end

  def unsubscribed
    # チャンネルからの切断時の処理（必要に応じて実装）
  end

  def speak(data)
    # クライアントから受信したメッセージをブロードキャスト
    ActionCable.server.broadcast("chat_channel", { message: data["message"] })
  end
end

File: app/javascript/channels/chat_channel.js
// app/javascript/channels/chat_channel.js
import consumer from "./consumer";

const chatChannel = consumer.subscriptions.create("ChatChannel", {
  connected() {
    console.log("Connected to ChatChannel");
  },
  disconnected() {
    console.log("Disconnected from ChatChannel");
  },
  received(data) {
    // 受信したメッセージをメッセージエリアに追加する
    const messagesDiv = document.getElementById("messages");
    const messageElement = document.createElement("div");
    messageElement.innerText = data.message;
    messagesDiv.appendChild(messageElement);
    // 最新メッセージが見えるようにスクロール
    messagesDiv.scrollTop = messagesDiv.scrollHeight;
  },
  speak(message) {
    this.perform("speak", { message: message });
  },
});

document.addEventListener("DOMContentLoaded", () => {
  const chatForm = document.getElementById("chat_form");
  const chatInput = document.getElementById("chat_input");

  chatForm.addEventListener("submit", (event) => {
    event.preventDefault();
    if (chatInput.value.trim() !== "") {
      chatChannel.speak(chatInput.value);
      chatInput.value = "";
    }
  });
});
